tango/integrations/beaker/executor.py

```diff
@@ -590,18 +592,32 @@ def _ensure_step_graph_dataset(self, step_graph: StepGraph) -> Dataset:
         return dataset
 
     def _ensure_cluster(self, task_resources: TaskResources) -> str:
-        cluster_to_use: str
+        cluster_to_use: Optional[str] = None
         if not self.clusters:
             raise ConfigurationError("At least one cluster is required in 'clusters'")
         elif len(self.clusters) == 1:
             cluster_to_use = self.clusters[0]
         else:
+
+            def recency_ranking(cluster_name: str):
+                try:
+                    return self._latest_clusters_used.index(cluster_name)
+                except ValueError:
+                    return -1
+
             available_clusters = sorted(
                 self.beaker.cluster.filter_available(task_resources, *self.clusters),
-                key=lambda x: x.queued_jobs,
+                key=lambda x: (x.queued_jobs, recency_ranking(x.cluster.full_name)),
```

Comment:
`queued_jobs` is the number of jobs queued but not running? Does it update every time you call it?

Reply:
Oops. Yes that does update every time

Reply:
It seems to me that this should sort by number of free slots first ... but I guess it's hard to find out what a "free slot" is given resource requirements.

As a stop-gap until Beaker has multi-cluster support, this is great.